### DISCUSSION TITLE
Inline loader seems not supporting loader option value in which type is function.

### DISCUSSION THREAD

**Discussion by mizok** [webpack/webpack#15362] [Q&A] [answered]
# Bug report

Here the doc mentioned that we have to pass a `JSON` or  a`query string` value for loader option when using ` inline loader `
https://webpack.js.org/concepts/loaders/#inline

But it seems both of `JSON` & `query string`  do not support  `function` values, so I suppose this is a bug? Since some loader does have options in which type is function. (for example: `html-loader` : `preprocessor`)

In my case, I just need `inline loader` to defined which loader to use for specific `.ejs` files, and I am going to use these files in `htmlWebpackPlugin` as template, so I can't use the `module.rules` way to set the loaders.

You can check the library I am working on here ( just read the `Passing individual values` part of `README.md`)
https://github.com/dc7290/template-ejs-loader

**What is the current behavior?**
If I pass a function type value as option value to `inline loader `, it just become `undefined`.

**If the current behavior is a bug, please provide the steps to reproduce.**
You can try to pass the option `preprocessor` of `html-loader` using  `inline loader`, then you will get what I am meaning about I guess.

**What is the expected behavior?**
`inline loader` must support option values in any type. 

**Other relevant information:**
webpack version:  5.66.0
Node.js version: v14.17.5
Operating System:  macOS Catalina
Additional tools:

**Comment by alexander-akait** [comment]
Not related to webpack, please open an issue in html-loader

**Comment by mizok** [comment]
> Not related to webpack, please open an issue in html-loader

@alexander-akait 
But obviously any loader that have option values in function type will get this problem I guess?

**Reply by alexander-akait** [reply]
Please provide example github repo with example how you want to use it

**Reply by mizok** [reply]
@alexander-akait 
ok sure.

**Comment by mizok** [comment, marked as answer]
@alexander-akait 

Here is my repo. 
You can just `npm run dev` to check the preview.
And the problem is at `webpack.config.ts` line 209, I have commented about what the problem is , hope you can get what I am talking about.
repo:
https://github.com/mizok/webpack-inline-loader-sample/
webpack.config.ts:
https://github.com/mizok/webpack-inline-loader-sample/blob/master/webpack.config.ts

**Reply by alexander-akait** [reply]
oh, `HtmlWebpackPlugin` was not design for this usage, it is not webpack problems, it was not design, maybe in this case better to write own plugin

**Reply by mizok** [reply]
But I think it makes sense that inline loader shall support function-typed option value..., since it is kinda usage of loaders? ... and some loaders DO have function-typed option value.

Maybe I am wrong, but I think a little bit change on the `parseQuery` function will do.

**Reply by alexander-akait** [reply]
It is just query, there are any values can be there

**Reply by alexander-akait** [reply]
Loader should handle it on own side, webpack doesn't know about them

**Reply by mizok** [reply]
OK..., actually I am not trying to offend.

So you mean options like `preprocessor` are actually not made for `inline loader` usage.

If we still want such feature(ex: interpolation), we should find another way which is able to fit `inline loader` usage, or just make another loader instead of using `html-loader`.

Is my understanding correct?

**Reply by alexander-akait** [reply]
Yes, we should improve `interpolation` for html-loader, if you have ideas, feel free to send PRs
